### ISSUE TITLE
Problems on generating the embeding in distributed setting for GraphSAGE

### ISSUE THREAD

**Issue by RocmFang** [dmlc/dgl#4092]
Hi, I met a problem on the generated embedding after training for the train_dist_unsupervised code, it can not continue to save the embedding file, it seem to be blocking between rank 0 and other clients, the process  is below as:

.......
[6]Epoch Time(s): 555.1406, sample: 311.8036, data copy: 0.2666, forward: 20.4729, backward: 221.6903, update: 0.8708, #seeds: 747610, #inputs: 0
hehehehe:
[7]Epoch Time(s): 555.1352, sample: 354.4214, data copy: 0.3818, forward: 18.4738, backward: 180.9850, update: 0.8406, #seeds: 747610, #inputs: 0
hehehehe:
[1]Epoch Time(s): 555.1426, sample: 383.2907, data copy: 3.0581, forward: 31.6548, backward: 136.1956, update: 0.8473, #seeds: 747611, #inputs: 0
hehehehe:
|V|=1138499, eval batch size: 100000
|V|=1138499, eval batch size: 100000
|V|=1138499, eval batch size: 100000
|V|=1138499, eval batch size: 100000
|V|=1138499, eval batch size: 100000
|V|=1138499, eval batch size: 100000
|V|=1138499, eval batch size: 100000
|V|=1138499, eval batch size: 100000
2it [00:00,  3.61it/s]
2it [00:00,  2.57it/s]
2it [00:00,  2.54it/s]
2it [00:00,  2.30it/s]
2it [00:01,  1.58it/s]
2it [00:02,  1.47s/it]

**Comment by RocmFang** [comment]
I used the train_dist_unsupervised.py to run my dataset.
This problem was caused by different python versions, some nodes with the python3.8, I reinstall them to python3.6, and it works now.
By the way, I found that this code just runs on a single NUMA socket? If I want to run it for two sockets in a machine, how to do it? Assume 4 machine with 8 sockets, I set the num_trainers and num_servers to 2, it just works on 4 partitions, not the 8 partitions.

**Comment by RocmFang** [comment]
> are you running this example directly: `https://github.com/dmlc/dgl/blob/master/examples/pytorch/graphsage/dist/train_dist_unsupervised.py`? which dgl version, platform are you using?

I used the train_dist_unsupervised.py to run my dataset.
This problem was caused by different python versions, some nodes with the python3.8, I reinstall them to python3.6, and it works now.
By the way, I found that this code just runs on a single NUMA socket? If I want to run it for two sockets in a machine, how to do it? Assume 4 machine with 8 sockets, I set the num_trainers and num_servers to 2, it just works on 4 partitions, not the 8 partitions.

**Comment by Rhett-Ying** [comment]
you're training on CPU? why do you want to control the socket? We do not support pin threads/workload onto specific socket. but you could try to increase `num_trainers` for faster train.
Usually when train on GPU, we set `num_trainers` as the num of GPUs on each machine which will utilize all the GPUs then.

**Comment by RocmFang** [comment]
Thanks, got it.  Since a machine has two CPUs, we want to split into 2 computing nodes to run the code. If I set ### num_trainers to 2 and ### num_omp_threads is equal to the cores of a socket, is this ok?

**Comment by Rhett-Ying** [comment]
yes, you could try though 2 cpus seems very limited as several threads will be created in the background for the RPC communications. 2 cores on each cpu?

**Comment by RocmFang** [comment]
Thanks, our machine equips 2 cpus with 36 cores.

**Comment by Rhett-Ying** [comment]
oh, then I think you try with larger `num_trainers` such as `8`, `16` to obtain faster train speed.

**Comment by RocmFang** [comment]
Thanks for your suggestions.
